src/js/pie.js

```diff
@@ -0,0 +1,200 @@
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            jQueryDomToD3Dom: {
+                funcName: "gpii.chartAuthoring.pieChart.pie.jQueryDomToD3Dom",
+                args: ["{arguments}.0"]
+            },
+            addD3Listeners: {
+                funcName: "gpii.chartAuthoring.pieChart.pie.addD3Listeners",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+            },
+            draw: {
+                funcName: "gpii.chartAuthoring.pieChart.pie.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    /**
+     * Convert jQuery DOM elements to D3 DOM elements.
+     * @param domElem - a jQuery DOM element or an array of jQuery DOM elements
+     */
+    gpii.chartAuthoring.pieChart.pie.jQueryDomToD3Dom = function (domElem) {
+        return d3.selectAll(domElem.toArray());
+    };
+
+    gpii.chartAuthoring.pieChart.pie.addD3Listeners = function (that, elem, eventName, listener) {
+        var d3Elem = that.jQueryDomToD3Dom(elem);
+        d3Elem.on(eventName, function (data, i) {
+            fluid.invokeGlobalFunction(listener, [data, i, that]);
+        });
+    };
+
+    gpii.chartAuthoring.pieChart.pie.draw = function (that) {
+        var svg = that.svg,
+            pie = that.pie,
+            arc = that.arc,
+            color = that.color,
+            dataSet = that.model.dataSet,
+            sliceSelector = that.options.classNames.slice,
+            textSelector = that.options.classNames.text,
+            animationDuration = that.options.pieOptions.animationDuration;
+
+        var paths = svg.selectAll("path")
+            .data(pie(dataSet));
+        var texts = svg.selectAll("text")
+            .data(pie(dataSet));
+
+        // Draw pie slices
+        paths.enter()
+            .append("path")
+            .attr({
+                "fill": function(d, i) {
+                    return color(i);
+                },
+                "d": arc,
+                "class": sliceSelector
+            })
+            .each(function (d) {
+                this._current = d;
+            });
+
+        paths.transition().duration(animationDuration).attrTween("d", function (a) {
+            var i = d3.interpolate(this._current, a);
+            this._current = i(0);
+            return function(t) {
+                return arc(i(t));
+            };
+        });
+
+        paths.exit().remove();
+
+        // Draw texts for pie slices
+        texts.enter()
+            .append("text")
+            .attr("transform", function(d) {
+                return "translate(" + arc.centroid(d) + ")";
+            })
+            .attr({
+                "text-anchor": "middle",
+                "class": textSelector
+            })
+            .text(function(d) {
+                return d.value;
+            });
+
+        texts.transition().duration(animationDuration).attr("transform", function(d) {
```

Comment:
This appears to duplicate the transform that was used in the original definition of the text position - what does it do?